src/IntelliTect.Coalesce/TypeDefinition/TypeViewModel.cs

```diff
@@ -288,12 +288,12 @@ public TypeViewModel PureType
             {
                 if (IsArray)
                 {
-                    return ArrayType;
+                    return ArrayType!;
                 }
 
                 if (IsGeneric && (IsCollection || IsNullable))
                 {
-                    return FirstTypeArgument;
+                    return FirstTypeArgument!;
```

Comment:
It appears that `IsCollection` returns true for any `IEnumerable` (not just `IEnumerable<T>`). Isn't it then possible for `FirstTypeArgument` to be null?

Reply:
That's actually a different bug - `IsCollection` should have been checking `IsA(typeof(IEnumerable<>))`.